src/v2/view-builder/utils/AuthenticatorUtil.js

```diff
@@ -25,11 +26,12 @@ const getButtonDataSeAttr = function (authenticator) {
 
 /* eslint complexity: [2, 19] */
 const getAuthenticatorData = function (authenticator, isVerifyAuthenticator) {
-  const authenticatorType = authenticator.authenticatorType || authenticator.factorType;
-  const key = _.isString(authenticatorType) ? authenticatorType.toLowerCase() : '';
+  // `factorType` metadata added by oin is not in appState, map them to authenticatorKey.
+  const authenticatorKey = authenticator.authenticatorKey || FactorTypeToAuthenticatorKeyMap[authenticator.factorType];
```

Comment:
we don't need to care about `factorType` hence `FactorTypeToAuthenticatorKeyMap` looks unnecessary.